devicetypes/smartthings/zigbee-multi-button.src/zigbee-multi-button.groovy

```diff
@@ -0,0 +1,226 @@
+		state.pressTime = now()
+		return event
+	}
+}
+
+def sendEventToChild(buttonNumber, event) {
+	String childDni = "${device.deviceNetworkId}:$buttonNumber"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	child?.sendEvent(event)
+}
+
+def getBatteryPercentageResult(rawValue) {
+	log.debug 'Battery'
+	def volts = rawValue / 10
+	if (volts > 3.0 || volts == 0 || rawValue == 0xFF) {
+		[:]
+	} else {
+		def result = [
+				name: 'battery'
+		]
+		def minVolts = 2.1
+		def maxVolts = 3.0
+		def pct = (volts - minVolts) / (maxVolts - minVolts)
+		result.value = Math.min(100, (int)(pct * 100))
+		def linkText = getLinkText(device)
+		result.descriptionText = "${linkText} battery was ${result.value}%"
+		createEvent(result)
+	}
+}
+
+def refresh() {
+	return zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, batteryVoltage) +
+			zigbee.enrollResponse()
+}
+
+def ping() {
+	refresh()
+}
+
+def configure() {
+	def bindings = getModelBindings(device.getDataValue("model"))
+	return zigbee.onOffConfig() +
+			zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, batteryVoltage, DataType.UINT8, 30, 21600, 0x01) +
+			zigbee.enrollResponse() +
+			zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, batteryVoltage) + bindings
+}
+
+def installed() {
+	initialize()
+	sendEvent(name: "button", value: "pushed", isStateChange: true)
+	if(childDevices) {
+		def event
+		for(def endpoint : 2..device.currentValue("numberOfButtons")) {
+			event = createEvent(name: "button", value: "pushed", isStateChange: true)
+			sendEventToChild(endpoint, event)
+		}
+	}
+}
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	def numberOfButtons = modelNumberOfButtons[device.getDataValue("model")]
+	sendEvent(name: "numberOfButtons", value: numberOfButtons, displayed: false)
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+	if(!childDevices) {
+		addChildButtons(numberOfButtons)
+	}
+}
+
+private addChildButtons(numberOfButtons) {
+	for(def endpoint : 2..numberOfButtons) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName[0..-2] + "${endpoint}"
+			addChildDevice("Child Button", childDni, device.getHub().getId(), [
+					completedSetup: true,
+					label         : componentLabel,
+					isComponent   : true,
+					componentName : "button$endpoint",
+					componentLabel: "Button $endpoint"
+			])
+		} catch(Exception e) {
+			log.debug "Exception: ${e}"
+		}
+	}
+}
+
+private getBatteryVoltage() { 0x0020 }
+private getHoldTime() { 1000 }
+private getButtonMap() {[
+		"3450-L" : [
+				"01" : 4,
+				"02" : 3,
+				"03" : 1,
+				"04" : 2
+		],
+		"3450-L2" : [
+				"01" : 4,
+				"02" : 3,
+				"03" : 1,
+				"04" : 2
+		]
+]}
+private getModelNumberOfButtons() {[
+		"3450-L" : 4,
+		"3450-L2" : 4
+]}
+private getModelBindings(model) {
+	def bindings = []
+	switch(model) {
```

Comment:
I would switch this instead on `modelNumberOfButtons()`.

Reply:
So, instead of switching on `modelNumberOfButtons()` I've decided to just iterate with number of buttons. If any future device would work in different way, we can adjust it in no time.